File: app/src/main/java/com/cxh/mvpart/rx/exception/ApiException.java
package com.cxh.mvpart.rx.exception;

/**
 * @author Hai (haigod7[at]gmail[dot]com)
 *         2016/5/31
 */
public class ApiException extends Exception {
    private int code;
    private String displayMessage;

    public ApiException(Throwable throwable, int code) {
        super(throwable);
        this.code = code;
    }

    public void setDisplayMessage(String displayMessage) {
        this.displayMessage = displayMessage;
    }

    public String getDisplayMessage() {
        return displayMessage;
    }

    public int getCode() {
        return code;
    }
}

File: app/src/main/java/com/cxh/mvpart/rx/exception/ApiException.kt
package com.cxh.mvpart.rx.exception

/**
 * @author Hai (haigod7[at]gmail[dot]com)
 *         2016/5/31
 */
class ApiException(throwable: Throwable, val code: Int) : Exception(throwable) {
    var displayMessage: String? = null
}
